crates/tauri-bundler/src/bundle/macos/app.rs

```diff
@@ -358,6 +360,159 @@ fn copy_framework_from(dest_dir: &Path, framework: &str, src_dir: &Path) -> crat
+      .args(["-L", &dest_bin_path.to_string_lossy()])
+      .output()
+      .expect("Failed to execute otool");
+
+    let dylib_dependency_output_str =
+      std::str::from_utf8(&dylib_dependency_output.stdout).expect("Invalid UTF-8 in otool output");
+    let mut has_dylib_changes = false;
+
+    log::info!("Fix linking for {}", dest_bin_path.display());
+
+    for framework in frameworks.iter() {
+      if framework.ends_with(".dylib") {
+        let dylib_src_path = PathBuf::from(framework);
+        let dylib_name = dylib_src_path
+          .file_name()
+          .expect("Couldn't get library filename");
+        let dylib_framework_path = framework_dir.join(dylib_name);
+
+        let new_name = format!("@rpath/{}", dylib_name.to_string_lossy());
+
+        // Check if the output contains the name of the dylib
+        let depends_on_dylib = dylib_dependency_output_str
+          .lines()
+          .any(|line| line.contains(dylib_name.to_string_lossy().as_ref()));
+        log::info!(
+          "Fix linking for {} depends on {:?} {}",
+          dest_bin_path.display(),
+          dylib_name,
+          depends_on_dylib
+        );
+        if depends_on_dylib {
+          // Set the install_name for the dylib itself
+          let status = Command::new("install_name_tool")
+            .args(["-id", &new_name, &dylib_framework_path.to_string_lossy()])
+            .status()
+            .expect("Failed to execute install_name_tool -id");
+
+          if !status.success() {
+            log::error!("Failed to update linking for {}", dest_bin_path.display());
+            return Err(crate::Error::GenericError(format!(
+              "Failed to update linking for {}",
+              dest_bin_path.display()
+            )));
+          }
+
+          let status = Command::new("install_name_tool")
+            .args([
+              "-change",
+              &framework,
+              &new_name,
+              &dest_bin_path.to_string_lossy(),
+            ])
+            .status()
+            .expect("Failed to execute install_name_tool -change");
+
+          if !status.success() {
+            log::error!("Failed to update linking for {}", dest_bin_path.display());
+            return Err(crate::Error::GenericError(format!(
+              "Failed to update linking for {}",
+              dest_bin_path.display()
+            )));
+          }
+          has_dylib_changes = true;
+        }
+      }
+    }
+
+    if has_dylib_changes {
+      let status = Command::new("otool")
+        .args(["-L", &dest_bin_path.to_string_lossy()])
+        .status()
+        .expect("Failed to execute otool");
+
+      if !status.success() {
+        log::error!("Failed to check linking for {}", dest_bin_path.display());
+        return Err(crate::Error::GenericError(format!(
+          "Failed to check linking for {}",
+          dest_bin_path.display()
+        )));
+      }
+    }
+  }
+
+  let status = Command::new("codesign")
```

Comment:
we have our own signing management via the `sign_paths` variable in this module, we should reuse it (maybe we don't even need to codesign here?)